### PR TITLE
feat: refactored adding support for RETL and Profiles

### PR DESCRIPTION
**Pull request by RanjeetMishra**
# Description

1. Refactored to have explicit operators for Profiles and Reverse ETL.
2. Updated ReadME.
3. Added version in headers for API requests.

# Testing
1. Tested with airflow.
```
*** Reading local file: /Users/ranjeetmishra/airflow/logs/dag_id=rudderstack-sample/run_id=scheduled__2024-10-03T13:01:39.570003+00:00/task_id=rs_trigger_sync/attempt=1.log
[2024-10-03, 13:06:50 UTC] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: rudderstack-sample.rs_trigger_sync scheduled__2024-10-03T13:01:39.570003+00:00 [queued]>
[2024-10-03, 13:06:50 UTC] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: rudderstack-sample.rs_trigger_sync scheduled__2024-10-03T13:01:39.570003+00:00 [queued]>
[2024-10-03, 13:06:50 UTC] {taskinstance.py:1282} INFO - 
--------------------------------------------------------------------------------
[2024-10-03, 13:06:50 UTC] {taskinstance.py:1283} INFO - Starting attempt 1 of 2
[2024-10-03, 13:06:50 UTC] {taskinstance.py:1284} INFO - 
--------------------------------------------------------------------------------
[2024-10-03, 13:06:50 UTC] {taskinstance.py:1303} INFO - Executing <Task(RudderstackRETLOperator): rs_trigger_sync> on 2024-10-03 13:01:39.570003+00:00
[2024-10-03, 13:06:50 UTC] {standard_task_runner.py:55} INFO - Started process 15951 to run task
[2024-10-03, 13:06:50 UTC] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'rudderstack-sample', 'rs_trigger_sync', 'scheduled__2024-10-03T13:01:39.570003+00:00', '--job-id', '29', '--raw', '--subdir', 'DAGS_FOLDER/retl_dag.py', '--cfg-path', '/var/folders/j3/0xg77yp13n79lpysgzml60780000gn/T/tmprlpdlbwc']
[2024-10-03, 13:06:50 UTC] {standard_task_runner.py:83} INFO - Job 29: Subtask rs_trigger_sync
[2024-10-03, 13:06:50 UTC] {task_command.py:388} INFO - Running <TaskInstance: rudderstack-sample.rs_trigger_sync scheduled__2024-10-03T13:01:39.570003+00:00 [running]> on host ranjeets-macbook-pro.local
[2024-10-03, 13:06:50 UTC] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=airflow@example.com
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=rudderstack-sample
AIRFLOW_CTX_TASK_ID=rs_trigger_sync
AIRFLOW_CTX_EXECUTION_DATE=2024-10-03T13:01:39.570003+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2024-10-03T13:01:39.570003+00:00
[2024-10-03, 13:06:50 UTC] {rudderstack.py:165} INFO - Triggering sync for retl connection id: 2cwRi0iuL5XMnnZCRj9BnPwUxau
[2024-10-03, 13:06:50 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:06:50 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:06:51 UTC] {rudderstack.py:65} INFO - poll and wait for sync to finish for retl-connectionId: 2cwRi0iuL5XMnnZCRj9BnPwUxau, syncId: crv9dqt52mjsic2pne00
[2024-10-03, 13:06:51 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:06:51 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:06:52 UTC] {rudderstack.py:196} INFO - Polled status for syncId: crv9dqt52mjsic2pne00 for retl connection: 2cwRi0iuL5XMnnZCRj9BnPwUxau, status: running
[2024-10-03, 13:07:02 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:07:02 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:07:03 UTC] {rudderstack.py:196} INFO - Polled status for syncId: crv9dqt52mjsic2pne00 for retl connection: 2cwRi0iuL5XMnnZCRj9BnPwUxau, status: running
[2024-10-03, 13:07:13 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:07:13 UTC] {base.py:73} INFO - Using connection ID 'rudderstack_default' for task execution.
[2024-10-03, 13:07:13 UTC] {rudderstack.py:196} INFO - Polled status for syncId: crv9dqt52mjsic2pne00 for retl connection: 2cwRi0iuL5XMnnZCRj9BnPwUxau, status: running

...
```

### CODE REVIEW

File: rudder_airflow_provider/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '1.1.0'
+__version__ = "1.1.0"
```
Comment (vyeshwanth) [inline rudder_airflow_provider/version.py:1]: @RanjeetMishra Are these changes intended?  If you are using any code formatter, it will nice to add a make command for the same

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+    "responses",
+    "setuptools"
+]
+requires-python = ">= 3.6"
+
+[tool.setuptools.packages.find]
+exclude = ["*test*"]
+
+[project.entry-points.apache_airflow_provider]
+provider_info = "sample_provider.__init__:get_provider_info"
```
Comment (vyeshwanth) [inline pyproject.toml:31]: Have you tested this?

Comment (RanjeetMishra) [reply pyproject.toml:31]: :D used AI and missed fixing this.

File: rudder_airflow_provider/version.py (hunk 1 above)
Comment (vyeshwanth) [reply rudder_airflow_provider/version.py]: we can just use pyproject.toml for now

**Review by vyeshwanth — APPROVED** [review]

File: rudder_airflow_provider/operators/rudderstack.py [hunk 3]
```diff
@@ -1,65 +1,119 @@
...
+            request_max_retries=self.request_max_retries,
+            poll_timeout=self.poll_timeout,
+            poll_interval=self.poll_interval,
+        )
+        profile_run_id = rs_profiles_hook.start_profile_run(self.profile_id)
         if self.wait_for_completion:
-            logging.info('waiting for sync to complete for retl-connecion: %s, syncId: %s', self.retl_connection_id, sync_id)
-            rs_hook.poll_retl_sync_status(self.retl_connection_id, sync_id)
\ No newline at end of file
+            logging.info(
```
Comment (RanjeetMishra) [inline rudder_airflow_provider/operators/rudderstack.py:116]: minor: logging -> self.log , check for other places as well

**Review by AchuthaSourabhC — APPROVED** [review]
